ansible_collections/arista/avd/plugins/action/yaml_templates_to_facts.py

```diff
@@ -37,6 +39,12 @@ def run(self, tmp=None, task_vars=None):
                     raise AnsibleActionFail("The argument 'templates' is not a list")
             else:
                 raise AnsibleActionFail("The argument 'templates' must be set")
+
+            output_filename = self._task.args.get("output_filename",False)
+            output_format = self._task.args.get("output_format","json")
+            template_output = self._task.args.get("template_output",False)
```

Comment:
Unable to understand the essence of this. So couldn't think of a ansible-test scenario for this.

Reply:
This becomes a bit "inception"-like. The point is if you have an input variable with a jinja2 expression, which cannot be resolved until after the data rendered in the template has been set.
Example:
Input variables:
```yaml
cvsourceip: "{{ management_interfaces.Vlan4092.ip_address | ipaddr('address') if type == 'l2leaf' else router_bgp.router_id }}"
```
Since `management_interfaces`and `router_bgp` are not set when the playbook is called, this jinja is just passed along raw. Leading to the output configuration containing the jinja2 expression instead of the intended value.

Today we work around this by writing the structured_config to a file using a template. This extra rendering step resolves the jinja expressions. Next task is importing the resulting file again, overwriting the facts set on the first run.

This new knob is a way of removing those two extra tasks from the role and running the extra templater inside the `yaml_templates_to_facts`. This makes it simpler to detect changes and setting the correct `changed` flag.

Reply:
I got it, you told this previously.